pep-0722.rst

```diff
@@ -0,0 +1,563 @@
+language, with Python scripts as a (better) alternative to shell scripts, batch
+files, etc. When used to create scripts, Python code is typically stored as a
+single file, often in a directory dedicated to such "utility scripts", which
+might be in a mix of languages with Python being only one possibility among
+many. Such scripts may be shared, often by something as simple as email, or a
+link to a URL such as a Github gist. But they are typically *not* "distributed"
+or "installed" as part of a normal workflow.
+
+One problem when using Python as a scripting language in this way is how to run
+the script in an environment that contains whatever third party dependencies are
+required by the script. There is currently no standard tool that addresses this
+issue, and this PEP does *not* attempt to define one. However, any tool that
+*does* address this issue will need to know what 3rd party dependencies a script
+requires. By defining a standard format for storing such data, existing tools,
+as well as any future tools, will be able to obtain that information without
+requiring users to include tool-specific metadata in their scripts.
+
+
+Rationale
+=========
+
+Because a key requirement is writing single-file scripts, and simple sharing by
+giving someone a copy of the script, the PEP defines a mechanism for embedding
+dependency data *within the script itself*, and not in an external file.
+
+We define the concept of a *metadata block* that contains information about a
+script. The only type of metadata defined here is dependency information, but
+making the concept general allows expansion in the future, should it be needed.
+
+In order to identify metadata blocks, the script can simply be read as a text
+file. This is deliberate, as Python syntax changes over time, so attempting to
+parse the script as Python code would require choosing a specific version of
+Python syntax. Also, it is likely that at least some tools will not be written
+in Python, and expecting them to implement a Python parser is too much of a
+burden.
+
+However, to avoid needing changes to core Python, the format is designed to
+appear as comments to the Python parser. It is possible to write code where a
+metadata block is *not* interpreted as a comment (for example, by embedding it
+in a Python multi-line string), but such uses are discouraged and can easily be
+avoided assuming you are not deliberately trying to create a pathological
+example.
+
+A `review <language survey_>`_ of how other languages allow scripts to specify
+their dependencies shows that a "structured comment" like this is a
+commonly-used approach.
+
+Specification
+=============
+
+Any Python script may contain one or more *metadata blocks*. Metadata blocks are
+identified by reading the script *as a text file* (i.e., the file is not parsed
+as Python source code), looking for contiguous blocks of lines that start with
+the identifying chacaters ``##``. Whitespace is not allowed before the
```

Comment:
Typo here, `chacaters`.

Reply:
Sigh. Thanks. Someday I'll see if I work out how to implement a Windows-compatible version of the `make spellcheck` command.

Reply:
`make spellcheck` didn't pick this one up, but to mirror the commands from the `Makefile`, it's essentially:

```sh
# Install
python3 -m pip install pre-commit

# Run
pre-commit run --all-files --hook-stage manual codespell
```

Or:
```sh
# Install
python3 -m pip install codespell

# Run, but it might pick up junk from .tox, .venv, other non-Git files
codespell

# Run on a file
codespell pep-0722.rst
```